catroid/src/org/catrobat/catroid/physics/PhysicsObject.java

```diff
@@ -333,6 +328,19 @@ protected void setCollisionBits(short categoryBits, short maskBits) {
 			filter.maskBits = maskBits;
 			fixture.setFilterData(filter);
 		}
+
+		if (updateState) {
+			updateNonCollidingState();
+		}
+	}
+
+	private void updateNonCollidingState() {
+		if (body.getUserData() != null && body.getUserData() instanceof Sprite) {
+			Object look = ((Sprite) body.getUserData()).look;
+			if (look != null && look instanceof PhysicsLook) {
+				((PhysicsLook) look).setNonColliding(collisionMaskRecord == PhysicsWorld.MASK_NOCOLLISION);
+			}
```

Comment:
For what it does: when collision bits of a physicsObject are changed, it passes this information  to the physicsLook's objectStateHandler which has some conditions depending on whether it is a colliding or non-colliding object. 

It is called mostly from the setType method and I couldn't find a better way of accessing the physicsObject's respective physicsLook.

What do you mean exactly by "passing it through the userdata"?